### ISSUE TITLE
formatters examples appear as html-encoded text

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#16676]
See https://eslint.org/docs/latest/user-guide/formatters/

For example, the `checkstyle` example used to look like this:

![image](https://user-images.githubusercontent.com/44349756/208209653-c7b300bd-e0db-4cd2-9de2-5e7b53b37939.png)

But now it looks like this:

![image](https://user-images.githubusercontent.com/44349756/208209596-fec773d9-599b-48dd-873c-88b0e2811603.png)

By checking deploy previews on PRs, it seems that this change was introduced by https://github.com/eslint/eslint/pull/16606

**Comment by amareshsm** [comment]
let me check this issue.

**Comment by chenxsan** [comment]
When using `text` language, we're actually [ opting-out of 11ty syntax highlighting](https://github.com/11ty/eleventy-plugin-syntaxhighlight/issues/54#issuecomment-1169031169). But I don't think that's the standard way to handle it.

Here's how GitHub renders `text` language https://github.com/eslint/eslint/blob/main/docs/src/user-guide/formatters/index.md:

![](https://user-images.githubusercontent.com/1091472/208293713-49355bdb-75f8-453f-88aa-39ab7a7fb6cb.png)

And this is how `markdown-it` renders it:

<img width="1185" alt="" src="https://user-images.githubusercontent.com/1091472/208294833-2d816b6f-2e5f-477c-a11c-2313ccca3ee1.png">

So I think it maybe more reasonable to decode the code block instead of relying on non-standard trick.

**Comment by mdjermanovic** [comment]
So, before https://github.com/eslint/eslint/pull/16606 contents of `text` blocks were not being html-escaped during the generation of html files from markdown, but now they are? In that case, it seems that we should skip html-escaping the formatters' output when generating the markdown file.

https://github.com/eslint/eslint/blob/e5ecfefa1c952195a3a8371f5953cc655d844079/templates/formatter-examples.md.ejs#L69

```diff
- <%= formatterResults[formatterName].result %>
+ <%- formatterResults[formatterName].result %>
```

**Comment by amareshsm** [comment]
As @chenxsan mentioned 

>So I think it maybe more reasonable to decode the code block in the markdown source code instead of hacking on the new syntax highlighter with non-standard trick. Another benefit would be more readable code block.

Can we do this? @mdjermanovic  thoughts?

@chenxsan  can we fix this issue as part of #16669  PR itself?

**Comment by chenxsan** [comment]
> So, before #16606 contents of `text` blocks were not being html-escaped during the generation of html files from markdown, but now they are?

Exactly, actually I can found some issues due to not being html-escaped before https://github.com/eslint/eslint/pull/16606. Here's a [preview page](https://deploy-preview-16605--docs-eslint.netlify.app/user-guide/configuring/ignoring-code#ignored-file-warnings) from https://github.com/eslint/eslint/pull/16605,

<img width="900" alt="" src="https://user-images.githubusercontent.com/1091472/209747544-f9e8c733-e47e-41c8-b67d-331df0ccc927.png">

And here's the [source code](https://github.com/eslint/eslint/blob/main/docs/src/user-guide/configuring/ignoring-code.md):

<img width="1318" alt="" src="https://user-images.githubusercontent.com/1091472/209747612-893b5d16-85f1-46dd-8a81-da2afc4e4e71.png">

As you can see, texts like `<relative/path/to/filename>` are missing in the rendered result before https://github.com/eslint/eslint/pull/16606.

And here's the [rendered version](https://deploy-preview-16606--docs-eslint.netlify.app/user-guide/configuring/ignoring-code#ignored-file-warnings) of #16606:

<img width="926" alt="CleanShot 2022-12-28 at 10 36 44@2x" src="https://user-images.githubusercontent.com/1091472/209748416-22383328-d9f7-44c2-9b5e-eaa691bb8067.png">


Back to this formatters' examples, I believe you're right. We no longer need to escape them here [formatter-examples.md.ejs](https://github.com/eslint/eslint/blob/e5ecfefa1c952195a3a8371f5953cc655d844079/templates/formatter-examples.md.ejs#L69). I've pushed a fix in https://github.com/eslint/eslint/pull/16669

**Comment by mdjermanovic** [comment]
> Back to this formatters' examples, I believe you're right. We no longer need to escape them here [formatter-examples.md.ejs](https://github.com/eslint/eslint/blob/e5ecfefa1c952195a3a8371f5953cc655d844079/templates/formatter-examples.md.ejs#L69). I've pushed a fix in #16669

Is this problem related to #16669? If not, it would be better to fix this in a separate PR.

**Comment by chenxsan** [comment]
> > Back to this formatters' examples, I believe you're right. We no longer need to escape them here [formatter-examples.md.ejs](https://github.com/eslint/eslint/blob/e5ecfefa1c952195a3a8371f5953cc655d844079/templates/formatter-examples.md.ejs#L69). I've pushed a fix in #16669
> 
> Is this problem related to #16669? If not, it would be better to fix this in a separate PR.

More like an issue should be caught and fixed in https://github.com/eslint/eslint/pull/16606, though #16669 is a follow-up of https://github.com/eslint/eslint/pull/16606. I think it's better to fixed in a separate PR.
